modules/commons/src/main/java/org/apache/synapse/commons/vfs/VFSUtils.java

```diff
@@ -270,7 +244,60 @@ public synchronized static boolean acquireLock(FileSystemManager fsManager, File
         }
         return false;
     }
-    
+
+    private static String getFullPath(FileObject fo) {
+        String fullPath = fo.getName().getURI();
+        int pos = fullPath.indexOf('?');
+        if (pos != -1) {
+            fullPath = fullPath.substring(0, pos);
+        }
+        return fullPath;
+    }
+
+    private static boolean createLockFile(byte[] lockValue, FileObject lockObject, String fullPath)
+            throws FileSystemException {
+        // write a lock file before starting of the processing, to ensure that the
+        // item is not processed by any other parties
+        lockObject.createFile();
+        OutputStream stream = lockObject.getContent().getOutputStream();
+        try {
+            stream.write(lockValue);
+            stream.flush();
+            stream.close();
```

Comment:
Shall we close the stream on finally block? Otherwise, this can lead to a resource leak when an exception occurring on write() or flush method

Reply:
fixed